akka-stream/src/main/scala/akka/stream/impl/SetupStage.scala

```diff
@@ -43,6 +43,7 @@ import akka.stream.stage.OutHandler
     override def preStart(): Unit = {
       try {
         val sink = factory(materializer, attributes)
+        setKeepGoing(true)
```

Comment:
@johanandren This is the fix, correct? Do I need to do anything else?

Reply:
No I think that should be it, I'm not quite sure exactly how the stage now stops though, that should now need to be explicit, perhaps add the `assertAllStagesStopped { test body }` to see that we are not now leaking running/incomplete streams

Reply:
You are right, this is a problem. Adding `assertAllStagesStopped` fails ...

Reply:
OK, I fixed it now. The trick was to implement `Sink.fromMaterializer` in terms of `Flow.fromMaterializer`. All tests now pass `assertAllStagesStopped`.